BridgeAppSDK/SBAProfileItem.swift

```diff
@@ -409,27 +428,43 @@ open class SBAUserDefaultsProfileItem: SBAProfileItemBase {
     }
 }
 
-open class SBAUserProfileItem: SBAProfileItemBase {
-    lazy open var user: SBAUser = {
-        return SBAUser.shared
-    }()
+open class SBAUserProfileItem: SBAKeychainProfileItem, SBANameDataSource {
     
     override open var value: Any? {
+        
         get {
-            // need to special-case accessing properties that are defined in protocol extensions
-            // because they're not available via KVC
-            switch SBAUserProtocolExtension(sourceKey) {
-            case SBAUserProtocolExtension.givenName:
-                return user.givenName
-            case SBAUserProtocolExtension.fullName:
-                return user.fullName
+            switch SBAProfileSourceKey(sourceKey) {
+            case SBAProfileSourceKey.preferredName:
+                // For the preferred name, if not a separate value stored, then return the given name
+                return super.value ?? self.storedValue(forKey: SBAProfileSourceKey.givenName.rawValue)
+                
+            case SBAProfileSourceKey.fullName:
+                // For the full name, if this class is used the it is assumed that the full name is 
+                // not editable and that the given/family name are set using one-to-one fields
+                return self.fullName
+                
             default:
-                return user.value(forKey: sourceKey)
+                return super.value
             }
         }
         
         set {
-            user.setValue(newValue, forKey: sourceKey)
+            if !readonly {
+                super.value = newValue
+            }
         }
     }
+    
+    override open var readonly: Bool {
+        return self.sourceKey == SBAProfileSourceKey.fullName.rawValue
```

Comment:
*givenName* is read/write.  *preferredName* is read/write but if nil, return given name.